From 3fbe43c9f577cadd6b5136fda2e6a6c0b4e0651e Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Fri, 18 Jun 2021 10:42:32 +0900
Subject: [PATCH] ksmbd: replace KSMBD_SHARE_CONFIG_PATH with inline function
Git-commit: 3fbe43c9f577cadd6b5136fda2e6a6c0b4e0651e
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


replace KSMBD_SHARE_CONFIG_PATH with inline function.

Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/ksmbd_server.h      | 18 +++++++++++-------
 fs/cifsd/mgmt/share_config.c |  2 +-
 2 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/fs/cifsd/ksmbd_server.h b/fs/cifsd/ksmbd_server.h
index a915ca5596dc..55b7602b79bd 100644
--- a/fs/cifsd/ksmbd_server.h
+++ b/fs/cifsd/ksmbd_server.h
@@ -90,13 +90,17 @@ struct ksmbd_share_config_response {
 };
 
 #define KSMBD_SHARE_CONFIG_VETO_LIST(s)	((s)->____payload)
-#define KSMBD_SHARE_CONFIG_PATH(s)				\
-	({							\
-		char *p = (s)->____payload;			\
-		if ((s)->veto_list_sz)				\
-			p += (s)->veto_list_sz + 1;		\
-		p;						\
-	 })
+
+static inline char *
+ksmbd_share_config_path(struct ksmbd_share_config_response *sc)
+{
+	char *p = sc->____payload;
+
+	if (sc->veto_list_sz)
+		p += sc->veto_list_sz + 1;
+
+	return p;
+}
 
 struct ksmbd_tree_connect_request {
 	__u32	handle;
diff --git a/fs/cifsd/mgmt/share_config.c b/fs/cifsd/mgmt/share_config.c
index fac6034b97a9..cb72d30f5b71 100644
--- a/fs/cifsd/mgmt/share_config.c
+++ b/fs/cifsd/mgmt/share_config.c
@@ -139,7 +139,7 @@ static struct ksmbd_share_config *share_config_request(char *name)
 	share->name = kstrdup(name, GFP_KERNEL);
 
 	if (!test_share_config_flag(share, KSMBD_SHARE_FLAG_PIPE)) {
-		share->path = kstrdup(KSMBD_SHARE_CONFIG_PATH(resp),
+		share->path = kstrdup(ksmbd_share_config_path(resp),
 				      GFP_KERNEL);
 		if (share->path)
 			share->path_sz = strlen(share->path);
-- 
2.33.0

